editor/src/components/titlebar/title-bar.tsx

```diff
@@ -0,0 +1,118 @@
+  <div
+    onClick={onClick}
+    style={{
+      padding: '4px 0',
+      margin: '0 20px 0 10px',
+      cursor: 'pointer',
+    }}
+  >
+    <MenuIcons.Smiangle />
+  </div>
+)
+
+interface ProjectTitleProps {}
+
+const ProjectTitle: React.FC<React.PropsWithChildren<ProjectTitleProps>> = ({ children }) => {
+  return <div style={{ fontWeight: 400, fontSize: 12, padding: '0 10px' }}>{children}</div>
+}
+
+const TitleBar: React.FC<React.PropsWithChildren<TitleBarProps>> = () => {
+  const { loginState, projectName, isCodeEditorVisible, isPreviewPaneVisible, dispatch } =
+    useEditorState(
+      (store) => ({
+        dispatch: store.dispatch,
+        loginState: store.userState.loginState,
+        projectName: store.editor.projectName,
+        isCodeEditorVisible: store.editor.interfaceDesigner.codePaneVisible,
+        isPreviewPaneVisible: store.editor.preview.visible,
+      }),
+      'TitleBar',
+    )
+
+  const onClickLoginNewTab = useCallback(() => {
+    window.open(auth0Url('auto-close'), '_blank')
+  }, [])
+
+  const toggleLeftPanel = useCallback(() => {
+    let actions: Array<EditorAction> = []
+    actions.push(togglePanel('leftmenu'))
+    dispatch(actions)
+  }, [dispatch])
+
+  const toggleCodeEditorVisible = React.useCallback(
+    () => dispatch([setPanelVisibility('codeEditor', !isCodeEditorVisible)]),
+    [dispatch, isCodeEditorVisible],
+  )
+
+  const togglePreviewPaneVisible = React.useCallback(
+    () => dispatch([setPanelVisibility('preview', !isPreviewPaneVisible)]),
+    [dispatch, isPreviewPaneVisible],
+  )
+
+  const loggedIn = loginState.type === 'LOGGED_IN'
+
+  return (
+    <SimpleFlexRow
+      style={{
+        flexGrow: 0,
+        minHeight: 40,
+        fontWeight: 600,
+        letterSpacing: 0.2,
+        alignItems: 'center',
+        justifyContent: 'space-between',
+      }}
+    >
+      <div style={{ display: 'flex', height: '100%' }}>
+        <AppLogo onClick={toggleLeftPanel} />
+        {loggedIn ? (
+          <>
+            <span>
+              <MenuTile
+                selected={isCodeEditorVisible}
+                icon={<LargerIcons.Code />}
+                onClick={toggleCodeEditorVisible}
+                size='large'
+              />
+            </span>
+            <span>
+              <MenuTile
+                selected={isPreviewPaneVisible}
+                icon={<LargerIcons.PreviewPane />}
+                onClick={togglePreviewPaneVisible}
+                size='large'
+              />
+            </span>
+          </>
+        ) : null}
+        {/* <FullHeightButton onClick={exportToGithub}>Export to Github</FullHeightButton> */}
+      </div>
+
+      <div>
+        <ProjectTitle>{projectName}</ProjectTitle>
+      </div>
+
+      <div style={{ display: 'flex', height: '100%' }}>
+        <TestMenu />
+        {/* <TextButton onClick={exportToGithub}>Fork</TextButton> */}
+        {loggedIn ? null : (
```

Comment:
the loginState has multiple statuses (eg COOKIES_OR_LOCALFORAGE_UNAVAILABLE) where showing the "Sign in to Save" button is not the adequate UI choice. I think we should at least put a TODO here to make sure we don't forget that this needs improvements, or create a second const that is explicitly about showing the Sign In button

Reply:
Noted. We have a bunch of states in notification-bar.tsx (not sure if it's exhaustive), which is still being used as well.